File: _정답코드/5주차/05_0이_되는_구간합/solution.cpp
#include <vector>
#include <unordered_map>
using namespace std;

int solution(vector<int> numbers){
    int n = numbers.size();
    unordered_map<int, int> sumCount;
    sumCount[numbers[0]] = 1;
    
    for (int i = 1; i < n; i++) {
        numbers[i] += numbers[i - 1];
        if (sumCount.find(numbers[i]) != sumCount.end()) {
            sumCount[numbers[i]]++;
        } else {
            sumCount[numbers[i]] = 1;
        }
    }
    
    int count = 0;
    for (auto& pair : sumCount) {
        int key = pair.first;
        int value = pair.second;
        if (key == 0) {
            count += value * (value + 1) / 2;
        } else {
            count += value * (value - 1) / 2;
        }
    }
    
    return count;
}

File: _정답코드/5주차/05_0이_되는_구간합/solution.py
def solution(numbers):
    n = len(numbers)
    sumCount = {}
    sumCount[numbers[0]] = 1
    
    for i in range(1, n):
        numbers[i] += numbers[i - 1]
        if numbers[i] in sumCount:
            sumCount[numbers[i]] += 1
        else:
            sumCount[numbers[i]] = 1
    
    count = 0
    for key in sumCount:
        value = sumCount[key]
        if key == 0:
            count += value * (value + 1) // 2
        else:
            count += value * (value - 1) // 2
    
    return count

